includes/admin/class-give-html-elements.php

```diff
@@ -226,10 +228,104 @@ public function forms_dropdown( $args = array() ) {
 		return $output;
 	}
 
-	/**
-	 * Donors Dropdown
-	 *
-	 * Renders an HTML Dropdown of all donors.
+    /**
+     * @unreleased
+     */
+    public function campaigns_dropdown($args = [])
+    {
+        $defaults = [
+            'name' => 'campaigns',
+            'id' => 'campaigns',
+            'class' => '',
+            'multiple' => false,
+            'selected' => 0,
+            'chosen' => false,
+            'number' => 30,
+            'placeholder' => esc_attr__('All Campaigns', 'give'),
+            'data' => [
+                'search-type' => 'campaign',
+            ],
+            'query_args' => [],
+        ];
+
+        $args = wp_parse_args($args, $defaults);
+
+        $campaigns_args = wp_parse_args(
+            $args['query_args'],
+            [
+                'orderby' => 'id',
+                'order' => 'DESC',
+                'per_page' => $args['number'],
+            ]
+        );
+
+        /**
+         * Filter the campaigns dropdown.
+         *
+         * @unreleased
+         *
+         * @param array $campaigns_args Arguments for campaigns_dropdown query.
+         *
+         * @return array Arguments for campaigns_dropdown query.
+         */
+        $campaigns_args = apply_filters('give_campaigns_dropdown_args', $campaigns_args);
+
+        $cache_key = Give_Cache::get_key('give_campaigns', $campaigns_args, false);
```

Comment:
Curious if this will do anything 🤔

Reply:
Yeah, I was wondering the same. I don't recall having anything in v3 structure making use of `Give_Cache`. I'm removing that part now.